.gitignore

```diff
@@ -11,3 +11,7 @@ build
 **/.vs
 **/bin
 **/obj
+
+# C# Telemetry Client Generator Related
+telemetry/csharp/buildtemp/**
```

Comment:
Nit: I think `**/buildtemp` and `**/Generated` are fine to add to the gitignore as a whole